pydantic/main.py

```diff
@@ -457,6 +457,11 @@ def update_forward_refs(cls, **localns: Any) -> None:
             if type(f.type_) == ForwardRef:
                 f.type_ = f.type_._evaluate(globalns, localns or None)  # type: ignore
                 f.prepare()
+            if f.sub_fields:
+                for sub_f in f.sub_fields:
```

Comment:
Now that I think about it, I'm not sure this will work with deeply nested sub-fields, like `Optional[Union[ForwardRef('Node'), Leaf]]`. Let me check it first.

Reply:
I tested with deeply nested `Union[Union[Tuple[ForwardRef('Node'), str], int], Leaf]` and in fact, this version didn't work.

I just re-implemented to handle deeply nested sub-fields with `ForwardRef`.